js-watcher/cache_collection.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"context"
+	"log"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type CacheCollection map[string]*Cache 
+
+func (c *CacheCollection) synchronize(ctx context.Context) {
+	doneCh := make(chan bool)
+	for namespace, cache := range *c {
+		log.Printf("Starting sync for %s\n", namespace)
+		go cache.synchronize(ctx, doneCh)
+	}
+	syncCount := 0
+	log.Println("Waiting on cache synchronization")
+	for {
```

Comment:
What about adding a timeout mechanism here in case of issues?

Reply:
👍 but I will just do `log.Fatal` on a timeout there. Or at most retry a few times and then eventually crash the server. There is no point in running the server if we cannot be confident that step completed successfully.

Reply:
Sure - it's more about catching the case where it ends up waiting (for whatever reason) and never errors and never enters operation...

Reply:
Added a timeout. It is configurable with a default value of 10 mins for all caches to sync.